vignettes/SEmIRD.Rmd

```diff
@@ -0,0 +1,316 @@
+---
+title: "SEmIRD: multiple exposed compartments"
+author: "Ben Lambert"
+output: rmarkdown::html_vignette
+bibliography: references.bib
+vignette: >
+  %\VignetteIndexEntry{SEmIRD}
+  %\VignetteEngine{knitr::knitr}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  message = FALSE,
+  comment = "#>"
+)
+```
+
+## Introduction
+This vignette introduces the `SEmIRD` model. This model is an extension of the [SEIRD model](https://github.com/Como-DTC-Collaboration/como-models/blob/main/vignettes/SEIRD.Rmd), which is described by the following ODEs:
+
+$$\frac{\text{d}S}{\text{d}t} = -\beta S I,$$
+$$\frac{\text{d}E}{\text{d}t} = \beta S I -\kappa E,$$
+$$\frac{\text{d}I}{\text{d}t} =\kappa E - (\gamma + \mu) I,$$
+$$\frac{\text{d}R}{\text{d}t} = \gamma I,$$
+$$\frac{\text{d}D}{\text{d}t} = \mu I.$$
+In this vignette, we are going to focus on the time period that individuals spend, on average, in the exposed compartment. This time period is known as the latent period.
+
+We can gain insight into how the flow out of the $E$ compartment works by imagining that we exhaust the infected individuals at a given time: $t=0$. The new equation then becomes:
+
+$$\frac{\text{d}E}{\text{d}t} = -\kappa E,$$
+Supposing there is a given number of infected individuals at $t=0$ represented by $E=E_0$, we can solve the above equation to yield an expression for the number of exposed over time:
+
+$$E(t) = E_0 \exp (- \kappa t)$$
+representing exponential decay, which we can plot assuming $E_0=1$ and $\kappa=0.5$:
+```{r, fig.height = 5, fig.width = 7, warning=FALSE}
+library(tidyverse)
+library(comomodels)
+
+# function to return exponential decay
+E <- function(E_0, kappa, t) {E_0 * exp(-kappa * t)}
+
+# make values to plot
+E_0 <- 1
+kappa <- 0.5
+t <- seq(0, 10, 0.01)
+E_vals <- E(E_0, kappa, t)
+
+tibble(t=t, E=E_vals) %>% 
+  ggplot(aes(x=t, y=E)) +
+  geom_line() +
+  xlab("time (days)")
+```
+
+We can also show the effect of changing $\kappa$: unsurprisingly, as $\kappa$ increases, the rate at which individuals become infectious increases and there is faster decay.
+```{r, fig.height = 5, fig.width = 7}
+kappa_1 <- 0.5
+kappa_2 <- 1
+kappa_3 <- 2
+E_1 <- E(E_0, kappa=kappa_1, t=t)
+E_2 <- E(E_0, kappa=kappa_2, t=t)
+E_3 <- E(E_0, kappa=kappa_3, t=t)
+tibble(t, E=E_1, kappa=kappa_1) %>% 
+  bind_rows(tibble(t, E=E_2, kappa=kappa_2)) %>% 
+  bind_rows(tibble(t, E=E_3, kappa=kappa_3)) %>% 
+  ggplot(aes(x=t, y=E, colour=as.factor(kappa))) +
+  geom_line() +
+  scale_color_brewer("kappa", palette = "Dark2") +
+  xlab("time (days)")
+```
+
+## Stochastic models and their means
+An alternative way to view these exponential decay curves is as probability distributions governing the time taken for an exposed individual to become infectious: i.e. the duration of the latent period.
```

Comment:
These seem to be covered almost the same in the SEIRD vignette already. Shall we just make a reference or simplify here?

Reply:
I have taken this out of the SEIRD model now, so it's all here. So I think we should keep.